### PR TITLE
wip: Add support for 3rd party reporters

### PR DESCRIPTION
**Pull request by BernardoSilva**
- [ ] allow developers to setup their custom reporters from config file

### CODE REVIEW

File: behave/configuration.py [hunk 1]
```diff
@@ -673,6 +677,17 @@ def setup_formats(self):
             for name, scoped_class_name in self.more_formatters.items():
                 _format_registry.register_as(name, scoped_class_name)
 
+    def setup_reports(self):
+        """Register more, user-defined reporters by name."""
+        print('current reporters:', self.reporters)
```
Comment (veryhappythings) [inline behave/configuration.py]: Is print a good choice here? Can we use a logger instead?

File: behave/reporter/_registry.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        raise ValueError(message)
+    module_name, class_name = parse_scoped_name(scoped_class_name)
+    reporter_module = load_module(module_name)
+    reporter_class = getattr(reporter_module, class_name, None)
+    if reporter_class is None:
+        raise ImportError("CLASS NOT FOUND: %s" % scoped_class_name)
+    return reporter_class
+
+
+def select_reporter_class(formatter_name):
```
Comment (veryhappythings) [inline behave/reporter/_registry.py]: `formatter_name` should be `reporter_name` I think? Same throughout the method

File: behave/configuration.py [hunk 3]
```diff
@@ -673,6 +677,17 @@ def setup_formats(self):
...
+        if self.more_reporters:
+            for name, scoped_class_name in self.more_reporters.items():
+                reporter = _report_registry.load_reporter_class(scoped_class_name)
+                if not _report_registry.is_formatter_valid(reporter):
```
Comment (veryhappythings) [inline behave/configuration.py]: This should be `is_reporter_valid`, I think?

**Comment by jenisys** [comment]
One extension-point for output-reports is enough and that is currently for formatters.
Therefore, I will not put any effort into making reporters also an extension-point.

HINT:
You can use your own reporter class and use an adapter-class to register it as formatter.

**Pull request closed without merging**